docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1401 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #44.        if T_sliding_mu is None or T_sliding_std is None:
</pre><p>If you want to improve your programming skills, you need to focus more on finding ways to remove branching logic </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
Thanks for the feedback!  

Just to confirm, since this a private function and, as you said, we are going to precompute them in the parent function, there is no need to let user not provide these values (and thus avoid the `if` here),

Reply:
Exactly! I am excited that you "got the other point behind my point" 😄 

When you make parameters "optional" in a function, you should really stop and ask "why"? Is there already a valid "real" use case for this (and not a hypothetical use case) for this parameter? If not, then keep the parameter is mandatory as long as you can. But if you actually have a real need for it to be optional then you can change it when you need it (or if it makes the code easier to maintain). It's almost a bit of "defensive programming" to keep functions "rigid".